src/Pages/SubsitesVirtualPage.php

```diff
@@ -223,33 +222,27 @@ public function validURLSegment()
     {
         $isValid = parent::validURLSegment();
 
+        $filters = [
+            'URLSegment' => $this->URLSegment,
+        ];
+
         // Veto the validation rules if its false. In this case, some logic
         // needs to be duplicated from parent to find out the exact reason the validation failed.
         if (!$isValid) {
-            $IDFilter = $this->ID ? "AND \"SiteTree\".\"ID\" <> $this->ID" : null;
-            $parentFilter = null;
+            // Exclude the current page from the filter
+            if ($this->exists()) {
+                $filters['ID:not'] = $this->ID;
+            }
```

Comment:
Minor point, but without this check, isn't `$this->ID` a `0` anyway? In which case `ID:not => 0` is perfectly fine?

Reply:
True, will update